streams/src/test/java/org/apache/kafka/test/KStreamTestDriver.java

```diff
@@ -120,6 +120,22 @@ public StateStore getStateStore(String name) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public <K, V> void forward(K key, V value, String childName) {
+        ProcessorNode thisNode = currNode;
+        try {
+            for (ProcessorNode childNode : (List<ProcessorNode<K, V>>) thisNode.children()) {
+                if (childNode.name().equals(childName)) {
+                    currNode = childNode;
```

Comment:
That is right, since Kafka Streams processor topology traversal is Depth First, when we finished one child route we need to go back to the next child route, and hence resetting the the currNode.